lib/src/middleware_extensions.dart

```diff
@@ -0,0 +1,14 @@
+import 'handler.dart';
+import 'middleware.dart';
+
+/// Extensions on [Middleware] to aid in composing [Middleware] and [Handlers].
+///
+/// These members can be used in place of [Pipeline].
+extension MiddlewareExtensions on Middleware {
+  /// Merges `this` and [other] into a new [Middleware].
+  Middleware addMiddleware(Middleware other) =>
+      (Handler handler) => this(other(handler));
```

Comment:
Would nesting this very deeply hurt performance compared to using `Pipeline`? I'm worried about introducing a convenience method if some power users are going to have to steer away from it.

Reply:
I think the perf will be *identical* to what exists now. Look at `Pipeline` – it's the same thing!

Reply:
https://github.com/dart-lang/shelf/blob/a16ea0356c3f73fb9e44973b6a52a06c5546fc75/lib/src/pipeline.dart#L42